### PR TITLE
UCP/TEST: Adaptive progress optimization.

### PR DESCRIPTION
**Pull request by yosefe**
addresses #1318 
This PR disables progress for unused uct_iface's, including auxiliary ifaces used only for wireup.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4178/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4184/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
This test seems to hang on 
```
[ RUN      ] udrc/test_ucp_peer_failure_2pairs.status_after_error/0
```
The test produces the error message
```
[1501076147.862921] [pod9:30506:5]     ucp_worker.c:296  UCX  ERROR Signaling wakeup failed: Bad file descriptor
```
This shows up in a lot of the other tests too.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4194/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4202/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov @shamisp @alex-mikheev pls review

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4206/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c [hunk 1]
```diff
@@ -329,18 +398,222 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
...
 
+static void ucp_worker_iface_activate(ucp_worker_iface_t *wiface)
+{
+    ucp_worker_h worker = wiface->worker;
+    ucs_status_t status;
+
+    ucs_trace("activate iface %p acount=%u", wiface->iface,
+              wiface->activate_count);
+
+   if (wiface->activate_count++ > 0) {
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: How about to add some prioritization logic basing on activate_count?

Comment (yosefe) [inline src/ucp/core/ucp_worker.c]: what do you mean? more progress priority for certain transports?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4211/ for details (Mellanox internal link).

File: src/ucp/core/ucp_worker.c (hunk 1 above)
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_worker.c]: yes

**Pull request merged**